### DISCUSSION TITLE
onnx-preview1

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#844] [Announcements]
<em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/onnx-preview1'>onnx-preview1</a>.</em>

**Comment by hooke007** [comment]
之前测试的时候，TRT构建引擎的时候，这可以明显缩短时间
--skipInference
builder_optimization_level=0

对紧凑架构似乎优化没有太大效率影响，我在2080上观察不到差别，可能需要弱点的gpu

关掉这些额外优化好像也可以
use_cuda_graph
use_cublas
use_cudnn

**Reply by Blinue** [reply]
我测试的时候 3 级和 5 级优化有明显区别。现在是 5 级优化，以后会提供选项来修改优化等级。我认为对于 TensorRT，性能应放在第一位，构建时间长点也是可以接受的，而且引擎会被缓存，一个模型通常只需要构建一次。

以后会在构建引擎/编译着色器时展示一个 UI，防止用户以为缩放失败了。

**Comment by BOYA1999** [comment]
牛

**Comment by jnbe23** [comment]
I am using the onnx build for real-time antialiasing. Some models I have created only allow for static onnx builds instead of the dynamic required for magpie. These models have a better image quality with acceptable speed. Would it be possible to add support for static models in the future?

**Comment by animeojisan** [comment]
Thanks for the great app! Are you considering an update to allow swapping or multiple combinations of onnx filters like shaders? I love this app so much, so please consider it.

**Reply by Blinue** [reply]
Thank you for your interest, I'm glad to hear this project has been helpful to you. These features will be included once the functionality is officially implemented, though it may take some time. The project has been on pause for a while, as maintaining it has consumed much of my free time, but I plan to resume updates soon.

**Reply by animeojisan** [reply]
Thank you for your time! I am very, very excited to hear that you are planning to update the application!

**Comment by animeojisan** [comment]
I'm sorry for bothering you during your busy time.
My RTX 5070 Ti doesn't seem to work with TensorRT. Are there any measures I can take to fix this? CUDA doesn't work either; only DirectML is functional.

**Reply by Blinue** [reply]
TensorRT requires an update to support the RTX 50 series. You can try replacing the DLLs, it might work if you're lucky. Otherwise, recompiling Magpie with the latest SDK is necessary.

**Reply by animeojisan** [reply]
CUDA is now working, but TensorRT still doesn't function. I plan to use DirectML for the time being. Thank you!

**Reply by Blinue** [reply]
I'm planning to release a new version with support for the RTX 50 series, thanks for the feedback!

**Reply by animeojisan** [reply]
Thank you so much for your prompt response! ONNX-preview1 is an indispensable app for me, and I truly appreciate your help!
